tests/manifests/test_configs_consistency.py

```diff
@@ -0,0 +1,524 @@
+        pass
+
+
+# A mounted secret will be the source of a mounted path for each secret key
+@dataclass(frozen=True)
+class MountedSecret(SourceOfMountedPaths):
+    data: dict[str, str] = field(default_factory=dict)
+    mount_point: str = field(default_factory=str)
+
+    @classmethod
+    def from_template(cls, template, volume_mount):
+        assert template["kind"] == "Secret"
+        # When secret data is empty, `data:` is None, so use `get_or_empty`
+        template_data = get_or_empty(template, "data")
+        if "subPath" in volume_mount:
+            return cls(
+                data={volume_mount["mountPath"].split("/")[-1]: template_data[volume_mount["subPath"]]},
+                mount_point="/".join(volume_mount["mountPath"].split("/")[:-1]),
+            )
+        else:
+            return cls(data=template_data, mount_point=volume_mount["mountPath"])
+
+    def get_mounted_paths(self) -> list[tuple[ParentMount, MountNode | None]]:
+        return [
+            (ParentMount(self.mount_point), MountNode(k, b64decode(v).decode("utf-8"))) for k, v in self.data.items()
+        ]
+
+
+# A mounted configmap will be the source of a mounted path for each configmap key
+@dataclass(frozen=True)
+class MountedConfigMap(SourceOfMountedPaths):
+    data: dict[str, str] = field(default_factory=dict)
+    mount_point: str = field(default_factory=str)
+
+    @classmethod
+    def from_template(cls, template, volume_mount):
+        assert template["kind"] == "ConfigMap"
+        # When secret data is empty, `data:` is None, so use `get_or_empty`
+        template_data = get_or_empty(template, "data")
+        if "subPath" in volume_mount:
+            return cls(
+                data={volume_mount["mountPath"].split("/")[-1]: template_data[volume_mount["subPath"]]},
+                mount_point="/".join(volume_mount["mountPath"].split("/")[:-1]),
+            )
+        else:
+            return cls(data=template_data, mount_point=volume_mount["mountPath"])
+
+    def get_mounted_paths(self) -> list[tuple[ParentMount, MountNode | None]]:
+        return [(ParentMount(self.mount_point), MountNode(k, v)) for k, v in self.data.items()]
+
+
+# A mounted rendered configmap will be the source of a mounted path
+# For each render-config-* container output
+@dataclass(frozen=True)
+class MountedRenderedConfigEmptyDir(SourceOfMountedPaths):
+    render_config_outputs: set[str] = field(default_factory=set)
+    mount_point: str = field(default_factory=str)
+
+    @classmethod
+    def from_workload_spec(cls, workload_spec, before_index, volume_mount):
+        outputs = []
+        # We look for render-config containers which ran before the current container
+        for container_spec in workload_spec["initContainers"][:before_index]:
+            if "render-config" in container_spec["name"]:
+                args = container_spec.get("args") or container_spec["command"][1:]
```

Comment:
I'd prefer to look at the container image and args rather than the name. We know `command` isn't going to be used for `matrix-tools` due to https://github.com/element-hq/ess-helm/pull/820